uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/PopupMediaItem.cpp

```diff
@@ -52,52 +61,49 @@ QString PopupMediaItem::title() const
 }
 
 /*!
-  \brief Returns the sourceUrl of the \c MediaPopupElement.
+  \brief Returns the caption of the \c MediaPopupElement.
  */
-QUrl PopupMediaItem::sourceUrl() const
+QString PopupMediaItem::caption() const
 {
-  return m_popupMedia->value()->sourceUrl();
+  return m_popupMedia->caption();
 }
 
 /*!
-  \brief Returns the linkUrl of the \c MediaPopupElement.
+  \brief Returns the popupMediaType of the \c MediaPopupElement.
  */
-QUrl PopupMediaItem::linkUrl() const
+PopupMediaType PopupMediaItem::popupMediaType() const
 {
-  return m_popupMedia->value()->linkUrl();
+  return m_popupMedia->popupMediaType();
 }
 
 /*!
-  \brief Returns the caption of the \c MediaPopupElement.
+  \brief Returns the PopupMedia of the \c MediaPopupElement.
  */
-QString PopupMediaItem::caption() const
+PopupMedia* PopupMediaItem::popupMediaItem() const
 {
-  return m_popupMedia->caption();
+  return m_popupMedia;
 }
 
-/*!
-  \brief Returns the popupMediaType of the \c MediaPopupElement.
- */
-PopupMediaType PopupMediaItem::popupMediaType() const
+QColor PopupMediaItem::jsonColorHelper(const QJsonArray& colorArray, const int index) const
```

Comment:
`const int` doesn't do much, but it does prevent you from changing `index` (even though it would have no side effects if you did). I've leaned towards just `int` rather than `const int`, but it does provide some value so let's go with what you have 👍